devlab/tests/test_verify_dst_functionality.py

```diff
@@ -0,0 +1,318 @@
+                msg = 'Cinder volume is OK and accessible on' \
+                      'VM with name {name} and ip: {addr}'
+                return msg.format(name=vm.name, addr=ip_addr), True
+            except SystemExit:
+                msg = 'Cinder volume access FAIL on' \
+                      'VM with name {name} and ip: {addr}'
+                return msg.format(name=vm.name, addr=ip_addr), False
+
+    def release_fips_tenant(self):
+        fip_tenant_ids = [f['id'] for f in
+                          self.dst_cloud.neutronclient.list_floatingips(
+        )['floatingips']
+            if f['tenant_id'] == self.dst_tenant_id]
+
+        for f_id in fip_tenant_ids:
+            self.dst_cloud.neutronclient.delete_floatingip(f_id)
+        time.sleep(3)
+
+    def update_floatip_neutron_quota(self, quota_value, tenant_name):
+        try:
+            # update tenant quota block
+            self.dst_cloud.switch_user(user=self.dst_cloud.username,
+                                       password=self.dst_cloud.password,
+                                       tenant='admin')
+            self.dst_cloud.neutronclient.update_quota(
+                self.dst_tenant_id, {'quota': {'floatingip': quota_value}})
+            self.dst_cloud.switch_user(user=self.dst_cloud.username,
+                                       password=self.dst_cloud.password,
+                                       tenant=tenant_name)
+        except (KeystoneClientBaseException, NovaClientBaseException,
+                CinderClientBaseException, GlanceClientBaseException,
+                NeutronClientException):
+            self.dst_cloud.switch_user(user=self.dst_cloud.username,
+                                       password=self.dst_cloud.password,
+                                       tenant=tenant_name)
+
+    def wait_service_on_vm_to_be_ready(self, timeout,
+                                       float_ip, service_port):
+        start_time = time.time()
+        end_time = start_time + timeout
+        nc_timeout_in_sec = 1
+        cmd_ssh_check = 'nc -zd -w {0} {1} {2}'.format(nc_timeout_in_sec,
+                                                       float_ip,
+                                                       service_port)
+        while end_time >= time.time():
+            with settings(warn_only=True):
+                result = local(cmd_ssh_check, capture=True)
+            if not result.failed:
+                return True
+            time.sleep(1)
+        return False
+
+    def wait_vm_ready(self, srv, timeout):
+        start_time = time.time()
+        end_time = start_time + timeout
+        while end_time >= time.time():
+            vm_status = self.dst_cloud.novaclient.servers.get(srv).status
+            if vm_status == 'ACTIVE':
+                pprint.pprint('VM is ready in {0}'
+                              'seconds'.format(time.time() - start_time))
+                return True
```

Comment:
Will be nice to handle case, when vm_status == 'ERROR'  - return False and do not wait 300 seconds